docs/products.md

```diff
@@ -0,0 +1,178 @@
+individually configured database which stores the analysis results. These
+subcommands are used to manage the products configured by the server. Please
+see the individual subcommands for details.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+available actions:
+  {list,add,del}
+    list                List products available on the server.
+    add                 Register a new product to the server.
+    del                 Delete a product from the server's products.
+
+common arguments:
+  --host HOST           The address of the CodeChecker server to connect to.
+                        (default: localhost)
+  -p PORT, --port PORT  The port the server is running on. (default: 8001)
+  --verbose {info,debug,debug_analyzer}
+                        Set verbosity level. (default: info)
+
+Most of these commands require authentication and appropriate access rights.
+Please see 'CodeChecker cmd login' to authenticate.
+~~~~~~~~~~~~~~~~~~~~~
+
+## Listing products (`list`)
+
+~~~~~~~~~~~~~~~~~~~~~
+usage: CodeChecker cmd products list [-h] [--host HOST] [-p PORT]
+                                     [-o {plaintext,rows,table,csv,json}]
+                                     [--verbose {info,debug,debug_analyzer}]
+
+List the name and basic information about products added to the server. The
+'S' status column of the product is 'E' if the product database is errorneous,
+and 'L' if you don't have access to the product.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+common arguments:
+  -o {plaintext,rows,table,csv,json}, --output {plaintext,rows,table,csv,json}
+                        The output format to use in showing the data.
+                        (default: plaintext)
+~~~~~~~~~~~~~~~~~~~~~
+
+## Adding new product (`add`)
+
+`add` assigns the unique `PRODUCT_NAME` endpoint with a database connection,
+making a new product available on the server.
+
+~~~~~~~~~~~~~~~~~~~~~
+usage: CodeChecker cmd products add [-h] [-n DISPLAY_NAME]
+                                    [--description DESCRIPTION]
+                                    [--sqlite SQLITE_FILE | --postgresql]
+                                    [--dbaddress DBADDRESS] [--dbport DBPORT]
+                                    [--dbusername DBUSERNAME]
+                                    [--dbpassword DBPASSWORD]
+                                    [--dbname DBNAME] [--host HOST] [-p PORT]
+                                    [--verbose {info,debug,debug_analyzer}]
+                                    PRODUCT_NAME
+
+Create a new product to be managed by the server by providing the product's
+details and database connection.
+
+positional arguments:
+  PRODUCT_NAME          The URL endpoint where clients can access the analysis
+                        results for this product.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -n DISPLAY_NAME, --name DISPLAY_NAME
```

Comment:
Access in the command-line is done through the endpoint (although `cmd products list` shows the displayed name). The display name is purely cosmetic, but the Web GUI filter filters based on display name. Think of it as run ID vs run name. Run name is purely cosmetic, as API access and database integrity is done through the ID.